contracts/libraries/Assessment/AssessmentIncidentsLib.sol

```diff
@@ -0,0 +1,129 @@
+  uint16 internal constant PERC_BASIS_POINTS = 10000;
+
+  // [todo] In case of duplicate incidents, allow an incident to be marked as duplicate by the
+  // proponent. They will need to provide an id which will compare productId, date, and priceBefore
+  // within certain tolerated ranges and if the two match, it allows the proponent to withdraw
+  // their deposit and transition the incident to a final state.
+
+  function releaseIncidentAssessmentDeposit (
+    uint104 id,
+    IAssessment.Incident[] storage incidents,
+    INXMToken nxm
+  ) external {
+    //IAssessment.Incident memory incident = incidents[id];
+
+    //require(block.timestamp >= incident.poll.end, "The incident is in cooldown period");
+
+    //uint16 assessmentDepositPerc = incident.details.assessmentDepositPerc;
+    //require(assessmentDepositPerc > 0, "Incident did not require an assessment deposit");
+
+    //IAssessment.PollStatus status = IAssessment._getPollStatus(incident.poll);
+    //uint payoutImpact = IAssessment._getPayoutImpactOfIncident(incident);
+    //uint deposit = payoutImpact * assessmentDepositPerc / PERC_BASIS_POINTS;
+
+    //require(incident.details.depositRedeemed, "Assessment deposit was already redeemed");
+    //incidents[id].details.depositRedeemed = true;
+    //if (status == IAssessment.PollStatus.ACCEPTED) {
+      //nxm.transferFrom(address(this), incidentProponent[id], deposit);
+    //}
+    //if (status == IAssessment.PollStatus.DENIED) {
+      //nxm.burn(deposit);
+    //}
+  }
+
+  function getIncidentToSubmit(
+    IAssessment.Configuration calldata CONFIG,
+    INXMToken nxm,
+    IMemberRoles memberRoles,
+    uint24 productId,
+    uint96 priceBefore,
+    uint32 date
+  ) external returns (IAssessment.AffectedToken memory, IAssessment.Incident memory) {
+    require(
+      memberRoles.checkRole(msg.sender, uint(IMemberRoles.Role.AdvisoryBoard)),
+      "Caller must be an advisory board member"
+    );
+    uint96 activeCoverAmount = 20000 ether; // NXM, since this will be driven by capacity
+    uint8 payoutAsset = uint8(IAssessment.Asset.ETH); // take this form product
+    address tokenAddress = 0x0000000000000000000000000000000000000000;
+
+    IAssessment.Incident memory incident = IAssessment.Incident(
+      IAssessment.Poll(0,0,uint32(block.timestamp), 0),
+      IAssessment.IncidentDetails(
+        productId,
+        date,
+        payoutAsset,
+        activeCoverAmount, // NXM
+        CONFIG.INCIDENT_ASSESSMENT_DEPOSIT_PERC,
+        CONFIG.INCIDENT_IMPACT_ESTIMATE_PERC,
+        false
+      )
+    );
+
+    uint payoutImpact = AssessmentUtilsLib._getPayoutImpactOfIncident(incident.details);
+    incident.poll.end = AssessmentUtilsLib._calculatePollEndDate(CONFIG, incident.poll, payoutImpact);
+
+
+    if (CONFIG.INCIDENT_ASSESSMENT_DEPOSIT_PERC > 0) {
+      uint deposit = payoutImpact * CONFIG.INCIDENT_ASSESSMENT_DEPOSIT_PERC / PERC_BASIS_POINTS;
+      nxm.transferFrom(msg.sender, address(this), deposit);
```

Comment:
the function is called getIncidentToSubmit but it has thiss side-effect here of sending NXM (not sure what is the reason for this one)

that's unexpected from a naming perspective

Reply:
True, I've updated this but for some reason github doesn't reflect the changes.